main.nf

```diff
@@ -166,8 +168,18 @@ workflow NFCORE_SAREK {
         aligner == "bwa-mem2" ? bwamem2 :
         dragmap
 
-    // TODO: add a params for msisensorpro_scan
-    msisensorpro_scan      = PREPARE_GENOME.out.msisensorpro_scan
+    // Reference msi list for MSIsensorpro
+    if (params.tools && params.tools.split(',').contains('msisensorpro')) {
+        if (params.msisensorpro_scan) {
+            msisensorpro_scan = Channel.fromPath(params.msisensorpro_scan).collect()
+        } else {
+            MSISENSORPRO_SCAN(fasta)
+            msisensorpro_scan = MSISENSORPRO_SCAN.out.list.map{ meta, list -> [list] }
+            versions = versions.mix(MSISENSORPRO_SCAN.out.versions)
```

Comment:
Is this something we can put in the PREPARE* subworkflows together with other reference computations for tools?

Reply:
I did that first, but it was being generated unnecessarily if msi analysis is not requested. I couldn't find a way to avoid that in the PREPARE* subworkflows.